### ISSUE TITLE
Lock not working with 1.70 and also seems to be an issue with refresh

### ISSUE THREAD

**Issue by Phoenix-DH** [itchannel/fordpass-ha#544]
Hey,

first thanks for working on that implementation.
Since yesterday we are owner of a Tourneo Custom 2024.
I installed according to the doc everything without any issue.

But what I figured out is:
Nothing happens when I use the lock. Nothing happens, also nothing in the logs.

And in general it seems to me that there is an refresh issue.

This is all I got related to the lock:
![IMG_9096](https://github.com/user-attachments/assets/01db67e0-f752-4d3a-81c1-1a8c859e3377)

**Comment by istbarp** [comment]
I have a focus 2019, and it seems lock is working fine with mine. 👌

![image](https://github.com/user-attachments/assets/01d443b7-ba93-455d-88fd-fcbe8ae01e57)

it doesn't always poll the key lock though. But if i lock via home assistant or app, it fetches status fine. Might be on the Ford side of things

**Comment by Phoenix-DH** [comment]
Hm okay,

that it might be a different behaviour when using HA or the exposed Homekit device like using the app I was also expecting. But after anperiod of time it shows always again unlocled and disarmed for the alarm.

Will maybe try a bit arround with some „initials“.

**Comment by Phoenix-DH** [comment]
I started to test the thibgs:

1.) FordPass App
Locking/Unlocking working, but no state reflection to HA. Nothing inside the logs.

2.) HA Button in the integration:
Stste label is always lock. Using it, it will always trigger a lock, regardless of the button stste. But I can see it at my car. In the logs it will always mentionen an unlock. Refresh time is current.

![IMG_9373](https://github.com/user-attachments/assets/41b2c99e-46d0-49ca-8964-7d44826ebb4b)
![IMG_9374](https://github.com/user-attachments/assets/9ff6f2ef-30c0-4d52-826a-b8b83e4e1a2d)
![IMG_9375](https://github.com/user-attachments/assets/e341c002-34d5-45b3-bfb7-5d2fa13845a9)

4.) Exposed homekit devices:
Always reflecting the state unlocked, same than in HA. Using it, do a lock. Button will switch to locked. But after the next sync state is back to unlocked.

So there might be something wrong with states. Maybe because api might be different with a 2024 modell, I dont know. I will also try to use the services with the next run.

Update:
What I just saw is thst now the last refresh was set 33 minutes ago. Integration is set to 900 seconds.

Action for poll api snd refresh will also do nothing. No state changes, no time updates.

Do I maybe have to enable something in Fordpass?

Update 2:
Using services inside ha for lock and locklockbis working completely fine for the unlock. But not for the lock. For rhe lock, the car was locked, but wrong action in HA as you can see in the screenshot. In both scenarios a unlock action seems to be triggered.
![IMG_9376](https://github.com/user-attachments/assets/cf7036aa-6a2d-4a59-a438-972e730d1d38)

**Comment by Cubixzno1** [comment]
I got the same "issues" as desbribed by you @Phoenix-DH - Did you ever find a fix?

**Comment by Phoenix-DH** [comment]
@Cubixzno1 
No, I played a lot arround as you can see in the big, but no solution.

@itchannel 
Thank you for the great work. Can we support here in some way to improve the implementatio?

**Comment by jb1228** [comment]
Hello @itchannel, I think I have a fix for this lock status issue. 

In the "**lock.py**" file, the `is_locked()` function is just getting the value of the _first_ element in the `doorLockStatus` array. 

``` python
return self.coordinator.data["metrics"]["doorLockStatus"][0]["value"] == "LOCKED"
```

I assume the first element is normally the status for "ALL_DOORS". However, my data looks something like this:

``` JSON
{
    "updateTime": "2025-04-03T00:00:00Z",
    "vehicleId": "REDACTED",
    "vin": "REDACTED",
    "metrics": {
        "doorLockStatus": [
            {
                "updateTime": "2025-04-03T00:00:00Z",
                "oemCorrelationId": "9999",
                "tags": {
                    "DOOR_LATCH_TYPE": "MECHANICAL"
                },
                "value": "NO_LOCKABLE_DEVICE",
                "vehicleDoor": "UNSPECIFIED_REAR",
                "vehicleOccupantRole": "PASSENGER",
                "vehicleSide": "DRIVER",
                "determinationMethod": "ACTUAL"
            },
            {
                "updateTime": "2025-04-03T00:00:00Z",
                "oemCorrelationId": "9999",
                "value": "LOCKED",
                "vehicleDoor": "ALL_DOORS",
                "determinationMethod": "ACTUAL"
            },
            {
                "updateTime": "2025-04-03T00:00:00Z",
                "oemCorrelationId": "9999",
                "tags": {
                    "DOOR_LATCH_TYPE": "MECHANICAL"
                },
                "value": "UNKNOWN",
                "vehicleDoor": "INNER_TAILGATE",
                "vehicleOccupantRole": "UNKNOWN",
                "vehicleSide": "UNKNOWN",
                "determinationMethod": "ACTUAL"
            },
            {
                "updateTime": "2025-04-03T00:00:00Z",
                "oemCorrelationId": "9999",
                "tags": {
                    "DOOR_LATCH_TYPE": "MECHANICAL"
                },
                "value": "NO_LOCKABLE_DEVICE",
                "vehicleDoor": "TAILGATE",
                "vehicleOccupantRole": "UNKNOWN",
                "vehicleSide": "UNKNOWN",
                "determinationMethod": "ACTUAL"
            },
            {
                "updateTime": "2025-04-03T00:00:00Z",
                "oemCorrelationId": "9999",
                "value": "LOCKED",
                "vehicleDoor": "UNSPECIFIED_FRONT",
                "vehicleOccupantRole": "DRIVER",
                "vehicleSide": "DRIVER",
                "determinationMethod": "ACTUAL"
            },
            {
                "updateTime": "2025-04-03T00:00:00Z",
                "oemCorrelationId": "9999",
                "tags": {
                    "DOOR_LATCH_TYPE": "MECHANICAL"
                },
                "value": "UNKNOWN",
                "vehicleDoor": "FRUNK",
                "vehicleOccupantRole": "UNKNOWN",
                "vehicleSide": "UNKNOWN",
                "determinationMethod": "ACTUAL"
            }
        ]
    }
}
```

<strike>The "ALL_DOORS" element is _second_ in the array. So changing the function to something like this _should_ work</strike>: 

Nevermind... see function updated [here](https://github.com/itchannel/fordpass-ha/blob/fb62c827b8264802047c0f58c6a00c7cc149a40d/custom_components/fordpass/lock.py#L63C1-L70C109) in the master branch.

**Comment by jb1228** [comment]
Actually... I see this was already fixed in #330 (with an added "DOUBLE_LOCKED" status check). The updated code is in the master branch, but not the 1.70 branch.

**Comment by Phoenix-DH** [comment]
@jb1228 

Than will mean if I exchange both files it may work? Is that the only one?

You will made my day.

**Comment by jb1228** [comment]
> Than will mean if I exchange both files it may work? Is that the only one?

@Phoenix-DH 

Correct! If you overwrite the `lock.py` file in the "1.70" branch with the one from the "master" branch, it _should_ work (after restarting Home Assistant). 

Just copy the whole file over as there are some other _minor_ updates in it. I don't see any other code in the repo that references the `doorLockStatus`, so that should be the only thing you need to change. 

Unfortunately, I cannot test it on my end since nothing in FordPass is working for me at the moment (even their app). Hopefully it is just temporary. 🤞

**Comment by Phoenix-DH** [comment]
@jb1228 @itchannel 

This really solved my issue! With the delay of 900 seconds, it is not that accurate when using the integration, but with the home integration in Apple it suits me. I tested all directions.

Thanks!
